### DISCUSSION TITLE
Validation feature

### DISCUSSION THREAD

**Discussion by kmurakami02** [cimug-org/CIMTool#169] [Q&A]
Hello everyone in the community, 

Does anyone have experience with validating a CIM grid model exported from OpenDSS with random a CIM profile? I am facing a problem below and seeking for a hint to overcome it.
I am doing this to learn how the validation function works and how the error messages highlight if there are mismatched between the model and profile. So, I am basically seeing a lot of messages within .diagnostic file, as I expected, however, some of the errors do not intuitively match to what I saw in the grid model and the profile. For example, an error message says:
`Undefined class CPSM-CoreEquipment:Terminal for uuid:B1506106-8CAE-4599-BCF8-B653DFBB1F30`
However, the profile has that class definition like XSD file shows:
![image](https://github.com/user-attachments/assets/8a938e02-2140-46fc-928d-3c248b68a2ce)
Then, the grid model seems to be based on the definition:
![image](https://github.com/user-attachments/assets/9d65009d-0800-4e31-9aff-48893da1eb17)
![image](https://github.com/user-attachments/assets/2ba17bba-e8a1-4bae-b939-d9c735afefb7)
In summary, the CIM-Tool apparently does not recognize the Terminal class instance as the instance is based on the class definition in the profile.

Any thoughts, comments, advise would be appreciated so much.

**Comment by tviegut** [comment]
@kmurakami02 :  Hi Kohei, I apologize for the delay!  Thanks for visiting the CIMTool discussion boards.  Your question might take a bit more time/space than an initial response here in a comment.  I sent an offline message to arrange for a bit more discussion.  I can add outcomes to this thread later on for anyone who might have similar questions.
